frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,256 @@
+ * Note that these metrics are only available on KFP v2 mode.
+ */
+export function MetricsTab({ execution }: MetricsTabProps) {
+  const [refreshCount, setRefreshCount] = useState(0);
+  function refresh() {
+    setRefreshCount(refreshCount + 1);
+  }
+  const [enableFetch, setEnableFetch] = useState(false);
+  const executionState = execution.getLastKnownState();
+  if (
+    !enableFetch &&
+    !(
+      executionState === Execution.State.NEW ||
+      executionState === Execution.State.UNKNOWN ||
+      executionState === Execution.State.RUNNING
+    )
+  ) {
+    setEnableFetch(true);
+  }
+
+  // Retrieving a list of artifacts associated with this execution,
+  // so we can find the artifact for system metrics from there.
+  const {
+    isLoading: isLoadingArtifacts,
+    isSuccess: isSuccessArtifacts,
+    error: errorArtifacts,
+    data: artifacts,
+  } = useQuery<Artifact[], Error>(
+    ['execution_output_artifact', { id: execution.getId() }],
+    () => getOutputArtifactsInExecution(execution),
+    { enabled: enableFetch, staleTime: Infinity },
+  );
+
+  // artifactTypes allows us to map from artifactIds to artifactTypeNames,
+  // so we can identify metrics artifact provided by system.
+  const {
+    isLoading: isLoadingArtifactTypes,
+    isSuccess: isSuccessArtifactTypes,
+    error: errorArtifactTypes,
+    data: artifactTypes,
+  } = useQuery<ArtifactType[], Error>(['artifact_types'], () => getArtifactTypes(), {
+    enabled: enableFetch,
+    staleTime: Infinity,
+  });
+
+  // This react element produces banner message if query to MLMD is pending or has error.
+  // Once query is completed, it shows actual content of metrics visualization in MetricsSwitcher.
+  return (
+    <ErrorBoundary>
+      <div className={commonCss.page}>
+        <div className={padding(20)}>
+          <div>
+            This step corresponds to execution{' '}
+            <Link
+              className={commonCss.link}
+              to={RoutePageFactory.executionDetails(execution.getId())}
+            >
+              "{ExecutionHelpers.getName(execution)}".
+            </Link>
+          </div>
+          {!enableFetch && <Banner message='Node has not completed.' mode='info' />}
+          {enableFetch && isLoadingArtifactTypes && isLoadingArtifacts && (
+            <Banner message='Metrics is loading.' mode='info' />
+          )}
+          {enableFetch && errorArtifacts && (
+            <Banner
+              message='Error in retrieving metrics information.'
+              mode='error'
+              additionalInfo={errorArtifacts.message}
+              refresh={refresh}
+            />
+          )}
+          {enableFetch && !errorArtifacts && errorArtifactTypes && (
+            <Banner
+              message='Error in retrieving artifact types information.'
+              mode='error'
+              additionalInfo={errorArtifactTypes.message}
+              refresh={refresh}
+            />
+          )}
```

Comment:
nit: same as above, probably no need to include `enableFetch`

Reply:
Sounds good, removed.